src/ripple/rpc/impl/RPCHelpers.cpp

```diff
@@ -651,28 +669,59 @@ keypairForSignature(Json::Value const& params, Json::Value& error)
                 std::string(jss::key_type) + " is used.");
             return { };
         }
-
-        seed = getSeedFromRPC (params, error);
     }
-    else
+
+    // ripple-lib encodes seed used to generate an Ed25519 wallet in a
+    // non-standard way. While we never encode seeds that way, we try
+    // to detect such keys to avoid user confusion.
     {
```

Comment:
We already know what field the secret is in. Lines 677-683 can be simplified to:
```
    if (secretType != jss::seed_hex.c_str()){
        seed = RPC::parseRippleLibSeed(params[secretType]);
```

Reply:
:+1: